src/main/kotlin/au/com/console/jpaspecificationdsl/JPASpecificationDSL.kt

```diff
@@ -23,6 +23,10 @@ private fun <T, R> KProperty1<T, R?>.spec(makePredicate: CriteriaBuilder.(path:
 fun <T, R> KProperty1<T, R?>.equal(x: R): Specifications<T> = spec { equal(it, x) }
 fun <T, R> KProperty1<T, R?>.notEqual(x: R): Specifications<T> = spec { notEqual(it, x) }
 
+fun <T, R: Any> KProperty1<T, R?>.`in`(x: Collection<R>): Specifications<T> = if (x.isNotEmpty()) spec {
```

Comment:
It's unfortunate that `in` is reserved, but we should be consistent with the Java API and keep it the same.

It's a bit hard to read this, maybe we could:

- rename `x` (we only used it for the other methods because that's what's used by the criteria API - but this has no corresponding method so we can call it whatever we like)
- introduce a `path` parameter for the spec lambda
- maybe a comment to document why this method isn't a pass-through to the criteria API